### PR TITLE
core, cli: use debug instead of raw console

### PR DESCRIPTION
**Pull request by JihangGuo**
[fix issue](https://github.com/alibaba/pipcook/issues/84)
1. use debug instead of raw console in cli/core
2. Modify some ESLint configurations([max-len](https://cn.eslint.org/docs/rules/max-len))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,4 +29,4 @@
     "@types/jasmine": "^3.5.10",
     "jasmine": "^3.5.0"
   }
-}
+}
```
Comment (yorkie) [inline package.json:32]: Missing EOF

File: .eslintrc.js [hunk 2]
```diff
@@ -27,23 +27,24 @@ module.exports = {
...
-    'arrow-spacing': [ 'error', { before: true, after: true } ],
+    'arrow-spacing': ['error', { before: true, after: true }],
     'block-spacing': 'error',
-    'brace-style': [ 'error', '1tbs', { allowSingleLine: true } ],
-    'comma-style': 'error'
+    'brace-style': ['error', '1tbs', { allowSingleLine: true }],
+    'comma-style': 'error',
+    'max-len': ["error", { "code": 80, "tabWidth": 2, "comments": 65 }]
   }
 };
```
Comment (yorkie) [inline .eslintrc.js:50]: These changed lines seem not related as the PR's title, may I ask you to revert these?

File: packages/cli/bin/index.js [hunk 3]
```diff
@@ -10,13 +10,16 @@ const start = require('../lib/start');
 const devPlugin = require('../lib/devPlugin');
 const dataset = require('../lib/dataset');
 const childProcess = require('child_process');
+const debugLog = require('debug')('cli/bin');
```
Comment (yorkie) [inline packages/cli/bin/index.js:13]: As for the debug's habit, use `debug` is better and simple:

```suggestion
const debug = require('debug')('cli/bin');
```

File: packages/cli/bin/index.js [hunk 4]
```diff
@@ -10,13 +10,16 @@ const start = require('../lib/start');
...
-  console.log(
+  debugLog(
     chalk.red(
-      `Pipcook requires node version higher than node 10.x. Howeverm your kicak node version is ${process.version}, ` +
-      'Please update node.js'
+      `Pipcook requires node version higher than node 10.x. 
+      Howeverm your kicak node version is 
+      ${process.version}
+      , Please update node.js`
     )
```
Comment (yorkie) [inline packages/cli/bin/index.js:23]: Debugging should not cover user-land logs, here it's used to tell users which node is older, so it's good to keep using log.

File: packages/cli/lib/board.js [hunk 5]
```diff
@@ -3,8 +3,8 @@ const fse = require('fs-extra');
 const ora = require('ora');
 const childProcess = require('child_process');
 const path = require('path');
-
 const spinner = ora();
+const debugLog = require('debug')('cli/board');
```
Comment (yorkie) [inline packages/cli/lib/board.js:7]: Ditto, using `debug` please :)

File: packages/cli/lib/board.js [hunk 6]
```diff
@@ -17,8 +17,7 @@ module.exports = () => {
       cwd: process.cwd(),
       stdio: 'inherit'
     });
-  
   } catch (e) {
-    console.error(chalk.red(e));
+    debugLog(chalk.red(e));
```
Comment (yorkie) [inline packages/cli/lib/board.js:21]: No necessary to use colors for debugging.

File: packages/cli/lib/dataset.js [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+const debugLog = require('debug')('cli/dataset');
 /**
  * prepare a working dir for developer to develop plugins
  */
 const dataset = async (cmdObj) => {
   const pluginType = cmdObj && cmdObj[0];
 
   if (!pluginType) {
-    console.log('Please provide a plugin type');
+    debugLog('Please provide a plugin type');
```
Comment (yorkie) [inline packages/cli/lib/dataset.js:12]: Ditto, this is for telling user's invalid input, this is not usecase for debug.

File: packages/cli/lib/devPlugin.js [hunk 8]
```diff
@@ -13,12 +14,19 @@ const devPlugin = (cmdObj) => {
   let projectName = cmdObj && cmdObj[1];
 
   if (!pluginType) {
-    console.log('Please provide a plugin type');
+    debugLog('Please provide a plugin type');
```
Comment (yorkie) [inline packages/cli/lib/devPlugin.js:17]: Ditto.

File: packages/cli/lib/devPlugin.js [hunk 9]
```diff
@@ -13,12 +14,19 @@ const devPlugin = (cmdObj) => {
...
     return;
   }
 
   if (!constants.PLUGINS.includes(pluginType)) {
-    console.warn(chalk.red(`Unsupported plugin type: "${pluginType}", it must be one of: \n${constants.PLUGINS.join(',\n')}`));
+    debugLog(
```
Comment (yorkie) [inline packages/cli/lib/devPlugin.js:22]: Ditto.

File: packages/core/src/utils/logger.ts [hunk 10]
```diff
@@ -5,6 +5,7 @@
 import chalk from 'chalk';
 import { PipcookRunner } from '../runner';
 import { PipcookComponentResult } from '../types/component';
+const debugLog = require('debug')('core/utils');
```
Comment (yorkie) [inline packages/core/src/utils/logger.ts:8]: As I mentioned before, debug and logger have different users, debug is for developer, and the log is for user.

**Comment by utkobe** [comment]
@JihangGuo Thanks for the contribution. May I ask you to sign the license/cla pls

**Comment by yorkie** [comment]
@JihangGuo BTW, it's better to have this("Modify some ESLint configurations") in another PR :p

**Comment by JihangGuo** [comment]
> @JihangGuo Thanks for the contribution. May I ask you to sign the license/cla pls

of course. done

**Comment by JihangGuo** [comment]
> @JihangGuo BTW, it's better to have this("Modify some ESLint configurations") in another PR :p

Thank you for your reply. I'll modification later😀

File: packages/core/src/config_test.ts [hunk 11]
```diff
@@ -1,8 +1,9 @@
 import config from './config';
+const debug = require('debug')('core/test');
```
Comment (yorkie) [inline packages/core/src/config_test.ts:2]:
```suggestion
const debug = require('debug')('pipcook/core/test');
```

File: packages/core/src/runner/helper.ts [hunk 12]
```diff
@@ -11,6 +11,8 @@ import { Observable, from } from 'rxjs';
 import { flatMap } from 'rxjs/operators';
 import { DATA, MODEL, EVALUATE, DEPLOYMENT, MODELTOSAVE } from '../constants/other';
 
+const debug = require('debug')('core/runner');
```
Comment (yorkie) [inline packages/core/src/runner/helper.ts:14]: See https://stackoverflow.com/questions/47724165/how-to-use-the-debug-module-with-typescript, change this to:

```ts
import Debug from 'debug';
const debug = Debug('pipcook/core/runner');
```

File: packages/core/package.json [hunk 13]
```diff
@@ -38,6 +38,7 @@
   "gitHead": "38b31171afad5051dc677d21a48e43dbc007b856",
   "devDependencies": {
     "@types/jasmine": "^3.5.7",
+    "debug": "^4.1.1",
```
Comment (yorkie) [inline packages/core/package.json:41]: And we need @types/debug, too.

**Comment by yorkie** [comment]
@JihangGuo Comments added, and there is a conflicting file, rebased with current master is required.

**Comment by yorkie** [comment]
Closing because of inactivity, feel free to reopen when you want to change again :)
